### DISCUSSION TITLE
Misp

### DISCUSSION THREAD

**Discussion by newmisp** [MISP/MISP#9743] [Q&A]
I installed MISP server but it cant be accessed via web

**Comment by iglocska** [comment]
This is a bit light on details...

**Comment by iglocska** [comment]
What distro? Which MISP installer did you use? Can you curl the web interface from the server itself? When you say the web interface can't be reached, what URL are you trying to use? Are you on the same network? Are you on the same machine? I need waaaaay more info than this.

**Comment by newmisp** [comment]
1. Ubuntu 22.04.4 LTS 
2. wget --no-cache -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
bash /tmp/INSTALL.sh
wget --no-cache -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
bash /tmp/INSTALL.sh -c
wget --no-cache -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
bash /tmp/INSTALL.sh -A
3. it is a ubuntu core no GUI 
4. I want to access it via its IP on the web
5. different network but connected via vpn

**Comment by iglocska** [comment]
if you curl localhost on the server itself, what do you get?

**Reply by newmisp** [reply]
it is a html document

**Comment by iglocska** [comment]
Does the html document contain this line?

```
    <title>Users - MISP</title>
```

**Reply by newmisp** [reply]
No

**Reply by newmisp** [reply]
Is there a way to uninstall misp ?

**Reply by iglocska** [reply]
Not really. Deleting the MISP directory / disabling + removing the apache config for MISP / dropping the MISP table in sql is the closest bet, but the requirements installed by the installer will stay.

**Reply by newmisp** [reply]
@iglocska Can you send me the exact commands to delete Delete the MISP directory / disabling + removing the apache config for MISP / dropping the MISP table in sql ?

**Reply by iglocska** [reply]
This might be a good idea to talk to your sysadmin if you are unsure about how to do those steps, but in case you are just experimenting with MISP on a non-production machine:

```
rm -rf /var/www/MISP
sudo mysql
drop database misp;
exit
sudo a2dissite misp-ssl
rm -f /etc/apache2/sites-available/misp-ssl.conf
```

**Reply by newmisp** [reply]
it is a test machine.

**Comment by newmisp** [comment]
Hi @iglocska . I installed MISP again on fresh Ubuntu 22.04 by the following commands:
wget --no-cache -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
bash /tmp/INSTALL.sh
wget --no-cache -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
bash /tmp/INSTALL.sh -A

but again the same problem, I cannot access it via webui. what you think about the possible issue with this ?

**Reply by HugeekMcGill** [reply]
Have you tried misp-docker (https://github.com/MISP/misp-docker) ? It will solve lots of your installation issues.

It saved me alot of time and headaches.

**Reply by newmisp** [reply]
No, I did not. after misp installation my VM has less storage which is why I might not able to try misp-docker

**Reply by HugeekMcGill** [reply]
I invite you to start a fresh VM, make a VM snapshoot (in case something goes wrong, you can come back to original fresh installtion)

Option 1 - Install from the sources
Option 2 - Use docker container images (Easier in my opinion)

**Comment by newmisp** [comment]


**Reply by HugeekMcGill** [reply]
Another question, if you are building from the sources, are you building from the main branch or a release tag ?

Main branch can still be in dev and release branches are more stable and predictable.

**Reply by newmisp** [reply]
I followed installation instructions from this page : https://misp.github.io/MISP/xINSTALL.ubuntu2204. I hope this answer your question.

**Reply by Rafiot** [reply]
You really need to send us the complete stacktrace from the command, there is no way we help when the only thing you tell us is "it's not working".

**Reply by Rafiot** [reply]
That's not what I'm asking for. When you run the install command, stuff should happen in the terminal. Somewhere in there, you will have the error message. 

We need to see these error messages before we can do anything more to help you.

**Reply by newmisp** [reply]
during the installation I saw only PHP not found error. that is it.

**Reply by Rafiot** [reply]
That will explain why nothing works. Yet, php is installed in that part of the script: https://misp.github.io/MISP/xINSTALL.ubuntu2204#2-install-lamp--dependencies

So you're going to have more messages than just that.

Please give us a screenshot or copy the last ~200 lines **of the terminal, not the browser**.

**Reply by newmisp** [reply]
I installed it yesterday. Is there a command to see the error logs during the installation ?

**Reply by Rafiot** [reply]
no, there is no such command.

In the future, when you install something and expect help, please make sure to document exactly what you do (every single command) and the every messages you're encountering in the terminal.

**Reply by Rafiot** [reply]
no, I can't. Not without a complete stacktrace.

**Reply by newmisp** [reply]
Hi @Rafiot. I have a fresh OS installed in my VM. Could you please give some advices for the MISP installation. For example, Is it OK if I follow https://misp.github.io/MISP/xINSTALL.ubuntu2204 for installation ? what will be the VM configuration like ? which ports should be allowed ? I tried to install it two times but it was not successful. your support will be appreciated.

**Reply by iglocska** [reply]
Are you trying to install it under ubuntu 22.04? The script should just work as is, php not being installed for you sounds like something fundamentally went wrong.

Either way, as @Rafiot described, we need the actual error messages to be able to help.

**Reply by newmisp** [reply]
yes, it is 22.04. during last installation I saw PHP not found error. later on, I found that port 80/443 were not allowed in the VM. did it effect the installation.?

**Reply by Rafiot** [reply]
I feel like we're in groundhog day. As we told you multiple times by now, we need detailed error messages and not just that one. The issue you're having could pretty much be anything from hardware to DNS issues.

**Reply by iglocska** [reply]
> I found that port 80/443 were not allowed in the VM.

Yes. If your VM cannot reach out via port 80/443, it cannot update anything / fetch requirements.

**Reply by newmisp** [reply]
I installed MISP again. this time, I can access it via Incognito private browsing but not from chromium or any other browsers. 

I got two errors :

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
vysion 1.0.10 requires pymisp==2.4.175, but you have pymisp 2.4.193 which is incompatible.

ERROR: setUpClass (__main__.TestComprehensive)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/www/MISP/PyMISP/tests/testlive_comprehensive.py", line 77, in setUpClass
    cls.admin_misp_connector = PyMISP(url, key, verifycert, debug=False)
  File "/var/www/MISP/PyMISP/pymisp/api.py", line 176, in __init__
    raise NoURL('Please provide the URL of your MISP instance.')
pymisp.exceptions.NoURL: Please provide the URL of your MISP instance.

----------------------------------------------------------------------
Ran 0 tests in 0.002s

FAILED (errors=1)
#########################################################################################  (88%)

**Reply by newmisp** [reply]
I also can access it via IP address in every browser

**Reply by newmisp** [reply]
Good morning @iglocska. I provide the correct base url for my MISP instance but still cannot access it via DNS. Do you have a solution for the issue or should I close this discussion ? thanks

**Comment by newmisp** [comment]
Closing the discussion because did not receive any response or assists.
